### ISSUE TITLE
group_data not available to @vagrant hosts, v0.14

### ISSUE THREAD

**Issue by Folo23** [Fizzadar/pyinfra#284]
**Describe the bug**
Variables declared in `group_data/{group_name or inventory file}.py` are not made available to `'@vagrant'` hosts. Only group variables declared in `group_data/all.py` are available to `@vagrant` hosts. Group variables appear to function correctly for regular hosts and those using the `@local` connector.

**To Reproduce**
```
$ vagrant init ubuntu/bionic64
$ vagrant up
$ echo "test_group = [\"@vagrant\"]" > test_inventory.py
$ mkdir group_data
$ echo "test_variable = \"hello\" > group_data/test_group.py
$ pyinfra test_inventory.py debug_inventory
--> Loading config...
--> Loading inventory...
    Getting Vagrant config...

--> Data for: @vagrant/default
{
    "ssh_hostname": "127.0.0.1",
    "ssh_port": "2200",
    "ssh_user": "vagrant",
    "ssh_key": "/home/frank/pyinfra_group_data_test/.vagrant/machines/default/virtualbox/private_key"
}
```

**Expected behavior**
```
$ pyinfra test_inventory.py debug_inventory
--> Loading config...
--> Loading inventory...
    Getting Vagrant config...

--> Data for: @vagrant/default
{
    "test_variable": "hello",
    "ssh_hostname": "127.0.0.1",
    "ssh_port": "2200",
    "ssh_user": "vagrant",
    "ssh_key": "/home/frank/pyinfra_group_data_test/.vagrant/machines/default/virtualbox/private_key"
}
```

**Meta**
+ Output of `pyinfra --support`:
```
--> Support information:

    If you are having issues with pyinfra or wish to make feature requests, please
    check out the GitHub issues at https://github.com/Fizzadar/pyinfra/issues .
    When adding an issue, be sure to include the following:

    System: Linux
      Platform: Linux-5.5.1-arch1-1-x86_64-with-glibc2.2.5
      Release: 5.5.1-arch1-1
      Machine: x86_64
    pyinfra: v0.14
    Executable: /home/frank/.local/share/virtualenvs/deployment-1aXbeu_h/bin/pyinfra
    Python: 3.8.1 (CPython, GCC 9.2.0)
```
+ Pyinfra was installed using pip, running in virtualenv.
+ Output when running with `--debug`:
```
$ pyinfra --debug test_inventory.py debug-inventory
    [pyinfra_cli.main] Checking potential directory: 
    [pyinfra_cli.main] Setting directory to: 
--> Loading config...
--> Loading inventory...
    [pyinfra_cli.inventory] Creating fake inventory...
    Getting Vagrant config...
[pyinfra.api.connectors.util] --> Waiting for exit status...
[pyinfra.api.connectors.util] --> Command exit status: 0
    [pyinfra.api.connectors.vagrant] Loading SSH config for default
[pyinfra.api.connectors.util] --> Waiting for exit status...
[pyinfra.api.connectors.util] --> Command exit status: 0
    [pyinfra.api.connectors.vagrant] Got Vagrant SSH info: 
['Host default', 'HostName 127.0.0.1', 'User vagrant', 'Port 2200', 'UserKnownHostsFile /dev/null', 'StrictHostKeyChecking no', 'PasswordAuthentication no', 'IdentityFile /home/frank/pyinfra_group_data_test/.vagrant/machines/default/virtualbox/private_key', 'IdentitiesOnly yes', 'LogLevel FATAL', '']
    [pyinfra_cli.inventory] Looking for group data in: group_data/test_group.py
    [pyinfra.api.connectors.vagrant] Got Vagrant SSH info: 
['Host default', 'HostName 127.0.0.1', 'User vagrant', 'Port 2200', 'UserKnownHostsFile /dev/null', 'StrictHostKeyChecking no', 'PasswordAuthentication no', 'IdentityFile /home/frank/pyinfra_group_data_test/.vagrant/machines/default/virtualbox/private_key', 'IdentitiesOnly yes', 'LogLevel FATAL', '']

--> Data for: @vagrant/default
{
    "ssh_hostname": "127.0.0.1",
    "ssh_port": "2200",
    "ssh_user": "vagrant",
    "ssh_key": "/home/frank/pyinfra_group_data_test/.vagrant/machines/default/virtualbox/private_key"
}
```
+ Output with `-vv` and `--debug`:
```
$ pyinfra -vv --debug test_inventory.py debug-inventory
    [pyinfra_cli.main] Checking potential directory: 
    [pyinfra_cli.main] Setting directory to: 
--> Loading config...
--> Loading inventory...
    [pyinfra_cli.inventory] Creating fake inventory...
    Getting Vagrant config...
localhost: >>> vagrant status --machine-readable
localhost: NOTE: Gem::Specification.default_specifications_dir is deprecated; use Gem.default_specifications_dir instead. It will be removed on or after 2020-02-01.                                                                                                                                              
localhost: Gem::Specification.default_specifications_dir called from /opt/vagrant/embedded/gems/gems/vagrant-2.2.7/lib/vagrant/bundler.rb:428.
localhost: NOTE: Gem::Specification.default_specifications_dir is deprecated; use Gem.default_specifications_dir instead. It will be removed on or after 2020-02-01.                                                                                                                                              
localhost: Gem::Specification.default_specifications_dir called from /opt/vagrant/embedded/gems/gems/vagrant-2.2.7/lib/vagrant/bundler.rb:428.
localhost: /opt/vagrant/embedded/gems/gems/vagrant-2.2.7/lib/vagrant/errors.rb:103: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call                                                                                                                    
localhost: /opt/vagrant/embedded/gems/gems/i18n-1.8.2/lib/i18n.rb:195: warning: The called method `t' is defined here
localhost: 1582153204,default,metadata,provider,virtualbox
localhost: 1582153204,default,provider-name,virtualbox
localhost: 1582153204,default,state,running
localhost: 1582153204,default,state-human-short,running
localhost: 1582153204,default,state-human-long,The VM is running. To stop this VM%!(VAGRANT_COMMA) you can run `vagrant halt` to\nshut it down forcefully%!(VAGRANT_COMMA) or you can run `vagrant suspend` to simply\nsuspend the virtual machine. In either case%!(VAGRANT_COMMA) to restart it again%!(VAGRANT_COMMA)\nsimply run `vagrant up`.
localhost: 1582153204,,ui,info,Current machine states:\n\ndefault                   running (virtualbox)\n\nThe VM is running. To stop this VM%!(VAGRANT_COMMA) you can run `vagrant halt` to\nshut it down forcefully%!(VAGRANT_COMMA) or you can run `vagrant suspend` to simply\nsuspend the virtual machine. In either case%!(VAGRANT_COMMA) to restart it again%!(VAGRANT_COMMA)\nsimply run `vagrant up`.
[pyinfra.api.connectors.util] --> Waiting for exit status...
[pyinfra.api.connectors.util] --> Command exit status: 0
    [pyinfra.api.connectors.vagrant] Loading SSH config for default
localhost: >>> vagrant ssh-config default
localhost: NOTE: Gem::Specification.default_specifications_dir is deprecated; use Gem.default_specifications_dir instead. It will be removed on or after 2020-02-01.                                                                                                                                              
localhost: Gem::Specification.default_specifications_dir called from /opt/vagrant/embedded/gems/gems/vagrant-2.2.7/lib/vagrant/bundler.rb:428.
localhost: NOTE: Gem::Specification.default_specifications_dir is deprecated; use Gem.default_specifications_dir instead. It will be removed on or after 2020-02-01.                                                                                                                                              
localhost: Gem::Specification.default_specifications_dir called from /opt/vagrant/embedded/gems/gems/vagrant-2.2.7/lib/vagrant/bundler.rb:428.
localhost: /opt/vagrant/embedded/gems/gems/vagrant-2.2.7/lib/vagrant/errors.rb:103: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call                                                                                                                    
localhost: /opt/vagrant/embedded/gems/gems/i18n-1.8.2/lib/i18n.rb:195: warning: The called method `t' is defined here
localhost: Host default
localhost: HostName 127.0.0.1
localhost: User vagrant
localhost: Port 2200
localhost: UserKnownHostsFile /dev/null
localhost: StrictHostKeyChecking no
localhost: PasswordAuthentication no
localhost: IdentityFile /home/frank/pyinfra_group_data_test/.vagrant/machines/default/virtualbox/private_key
localhost: IdentitiesOnly yes
localhost: LogLevel FATAL
localhost: 
[pyinfra.api.connectors.util] --> Waiting for exit status...
[pyinfra.api.connectors.util] --> Command exit status: 0
    [pyinfra.api.connectors.vagrant] Got Vagrant SSH info: 
['Host default', 'HostName 127.0.0.1', 'User vagrant', 'Port 2200', 'UserKnownHostsFile /dev/null', 'StrictHostKeyChecking no', 'PasswordAuthentication no', 'IdentityFile /home/frank/pyinfra_group_data_test/.vagrant/machines/default/virtualbox/private_key', 'IdentitiesOnly yes', 'LogLevel FATAL', '']
    [pyinfra_cli.inventory] Looking for group data in: group_data/test_group.py
    [pyinfra.api.connectors.vagrant] Got Vagrant SSH info: 
['Host default', 'HostName 127.0.0.1', 'User vagrant', 'Port 2200', 'UserKnownHostsFile /dev/null', 'StrictHostKeyChecking no', 'PasswordAuthentication no', 'IdentityFile /home/frank/pyinfra_group_data_test/.vagrant/machines/default/virtualbox/private_key', 'IdentitiesOnly yes', 'LogLevel FATAL', '']

--> Data for: @vagrant/default
{
    "ssh_hostname": "127.0.0.1",
    "ssh_port": "2200",
    "ssh_user": "vagrant",
    "ssh_key": "/home/frank/pyinfra_group_data_test/.vagrant/machines/default/virtualbox/private_key"
}
```

I wasn't able to figure out any further what was happening. Let me know if there's anything else I can do to help debug this. Been very much enjoying using pyinfra the past few days.

**Comment by Fizzadar** [comment]
@Folo23 glad to hear you're enjoying pyinfra! 

I've just identified and pushed a fix onto master for this issue (https://github.com/Fizzadar/pyinfra/commit/427ce02cdbe3a965bb4f8028700156775e1e7e4c) which I'll release in `v0.14.1` later today!

**Comment by Fizzadar** [comment]
Just released `v0.14.1` with this :)

**Comment by Folo23** [comment]
Thanks @Fizzadar, that fixed it!
